src/watch/index.ts

```diff
@@ -98,6 +86,25 @@ export const watchCommand = command({
 				}
 			}
 		});
+	};
+
+	const reRun = debounce(() => {
+		if (
+			runProcess
+			&& (!runProcess.killed && runProcess.exitCode === null)
+		) {
+			runProcess.kill();
+		}
```

Comment:
Can you move this logic down right after the `exit` handler is registered?

Reply:
@privatenumber like this?

Reply:
@privatenumber, could you take a look? Fixing this problem would improve developer experience quite a lot

Reply:
it is a very useful feature, PTAL @privatenumber